src/test/ui/consts/recursive-zst-static.rs

```diff
@@ -1,6 +1,4 @@
-// build-pass
-
-static FOO: () = FOO;
+static FOO: () = FOO; //~ cycle detected when const-evaluating `FOO`
```

Comment:
Please add a comment explaining why this failure is important, linking to the issue this fixes.

Reply:
done